### DISCUSSION TITLE
Correct way to caputre a validator exception and return HTTPException

### DISCUSSION THREAD

**Discussion by brymon68** [pydantic/pydantic#4561] [Question] [answered]
I am using FastAPI with Pydantic. I would like to be able to capture the raised ValueError from my validator and return an HTTPException to the client that I control. Is this possible? Something like: 

```
from typing import List, Optional

from pydantic import BaseModel, validator


class Query(BaseModel):
    foo: str,
    bar: str,

    @validator("foo")
    def validate_foo(cls, v, values):
       if v == "blah":
            raise ValueError("Cant have value blah")
        return v

```

And my router: 
```
@router.post("/foo")
def create_query(request: Request, item: Query):
  

   # if something bad happens in validator, catch error and raise HTTPException
   raise HTTPException(status_code=400, detail="Cant have value blah")
 ```

**Comment by samuelcolvin** [comment, marked as answer]
Yes I think it's possible, but it's a question for fastapi, not really pydantic.

Search for "customizing exceptions with fastapi" or similar.

**Reply by brymon68** [reply]
Sweet. Thanks @samuelcolvin. Following: https://fastapi.tiangolo.com/tutorial/handling-errors/
